pym/bob/input.py

```diff
@@ -2373,6 +2373,10 @@ def __parse(self):
 
         # finally parse recipes
         if not os.path.isdir("recipes"):
+            self.__cache.close()
```

Comment:
No, the exception would be thrown before the try/finally is entered. I mean by moving

```python
        # finally parse recipes
        if not os.path.isdir("recipes"):
            raise ParseError("No recipes directory found.")
```

above `self.__cache.open()` will not create the file at all.